src/App.jsx

```diff
@@ -1,10 +1,100 @@
-function App() {
+import { useEffect, useState, useMemo } from 'react';
+import { Container, Box, Typography, Button } from '@mui/material';
+import { SearchBar } from './components/SearchBar/SearchBar';
+import { ArticlesList } from './components/ArticlesList/ArticlesList';
+import { ArticleModal } from './components/ArticleModal/ArticleModal';
+import { SortButton } from './components/SortButton/SortButton';
+
+const App = () => {
+  const [articles, setArticles] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [open, setOpen] = useState(false);
+  const [sortOrder, setSortOrder] = useState('none'); // 'none', 'ascending', 'descending'
+  const [favoriteArticles, setFavoriteArticles] = useState(() => {
+    const saved = localStorage.getItem('favoriteArticles');
+    return saved ? JSON.parse(saved) : [];
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const [articlesResponse] = await Promise.all([
+          fetch('http://localhost:3010/articles'),
+        ]);
+        const articlesData = await articlesResponse.json();
+        setArticles(articlesData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [articles]);
+
+  useEffect(() => {
+    localStorage.setItem('favoriteArticles', JSON.stringify(favoriteArticles));
+  }, [favoriteArticles]);
+
+  const handleToggleFavorite = (articleId) => {
+    setFavoriteArticles((previous) => {
+      if (previous.includes(articleId)) {
+        return previous.filter((id) => id !== articleId);
+      }
+      return [...previous, articleId];
+    });
+  };
+
+  const filteredAndSortedArticles = useMemo(() => {
+    let result = articles.filter(
+      (article) =>
+        article.title.toLowerCase().includes(searchQuery.toLowerCase()) ||
+        article.content.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+
+    if (sortOrder !== 'none') {
```

Comment:
This is achievable with a query parameter.
http://localhost:3000/api#/default/ArticlesController_getAll

![image](https://github.com/user-attachments/assets/bd5eec97-4e3c-4368-baa6-3ecd060e19c2).

By the way, you shouldn't use `let` here. Instead, you could return the sorted articles in this `if` statement